clients/gitlabrepo/checkruns.go

```diff
@@ -32,11 +35,16 @@ func (handler *checkrunsHandler) init(repourl *repoURL) {
 }
 
 func (handler *checkrunsHandler) listCheckRunsForRef(ref string) ([]clients.CheckRun, error) {
-	pipelines, _, err := handler.glClient.Pipelines.ListProjectPipelines(
-		handler.repourl.projectID, &gitlab.ListProjectPipelinesOptions{
-			SHA:         &ref,
-			ListOptions: gitlab.ListOptions{},
-		})
+	var options gitlab.ListProjectPipelinesOptions
+
+	// In Gitlab: ref refers to a branch or tag name, and sha refers to commit sha
+	if gitCommitHashRegex.MatchString(ref) {
+		options.SHA = &ref
+	} else {
+		options.Ref = &ref
```

Comment:
Does options.Ref also work if the user passes in something like a 7-character SHA?

Reply:
So initially I set it up to only count full length shas, since thats what is pulled in the `ListCommits()` function, even if a user passes in the shortsha, tag or branch with the `--commit` flag. I hadn't considered to try if the shortsha works in the ref param. 

Testing with postman against the [ListProjectPipelines](https://docs.gitlab.com/ee/api/pipelines.html#list-project-pipelines) endpoint:

- Sha Param: the full length Sha returns the expected commits, the short sha (8 characters in gitlab, I also tried 7) returns an empty list. 
- Ref Param: Branch name returns the expected commits, tags and shas (40 or 8 characters) returns an empty list. Given the documentation I expected tags to also work. 

I'll have to update the comment on 40 to reflect this.

Reply:
That's weird about the tags. Can always fix it later if we find out we're doing something wrong.

e.g. if we needed to use `tags/foo` instead of `foo` (i tried a few variations and couldn't get it to work)